src/test/java/com/cognifide/aemrules/htl/checks/AbstractHtlCheckTest.java

```diff
@@ -0,0 +1,58 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015-2018 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl.checks;
+
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.File;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.plugins.html.checks.HtmlIssue;
+import org.sonar.plugins.html.visitor.HtmlSourceCode;
+
+public class AbstractHtlCheckTest {
+
+    private AbstractHtlCheck check;
+
+    @Before
+    public void setUp() {
+        this.check = new AbstractHtlCheck();
+    }
+
+    @Test
+    public void checkIssues() {
```

Comment:
Please divide these test into tests:
creatingViolation without cost (line nr 0 and different number)
creatingViolation with cost (line nr 0 nad different line number)
creatingViolation with cost and without (same like this test)